### PR TITLE
[RFR] gh-actions: bye bye travis \o

### PR DESCRIPTION
**Pull request by digitronik**
- Moving CI from travis to gh-actions
- Added python 3.8 test support
- Added cron job run so it will run on every Friday
- Added basic pre-commit (not black); Open the small window so we can adopt other hooks in the future. 
- Apply some end-line fixes with pre-commit

Run:
https://github.com/digitronik/widgetastic.core/actions/runs/235811401

coverage report:
https://codecov.io/gh/digitronik/widgetastic.core/branch/gh_action

I will update begets as per status.

### CODE REVIEW

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    - name: Checkout to master
+      uses: actions/checkout@master
+
+    - name: Set up Python-${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Set up Selenium Container
+      run: docker run -d --network host -p 4444:4444 -v /dev/shm:/dev/shm quay.io/redhatqe/selenium-standalone
```
Comment (quarckster) [inline .github/workflows/test_suite.yml:54]: Let's use `podman` instead:
```
podman run -d --network host -p 4444:4444 --shm-size=2g quay.io/redhatqe/selenium-standalone:latest
```

File: setup.cfg [hunk 2]
```diff
@@ -15,6 +15,7 @@ classifiers =
     Programming Language :: Python :: 3.5
```
Comment (quarckster) [inline setup.cfg:15]: I guess we can remove this line when excluded Python 3.5 from testing matrix

File: tox.ini [hunk 3]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{36,37},pypy3,codechecks
+envlist = py{35,36,37,38},pypy3,codechecks
```
Comment (quarckster) [inline tox.ini:2]: Let's drop py35

File: tox.ini [hunk 4]
```diff
@@ -1,5 +1,5 @@
...
 
 [testenv]
 passenv = TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH BROWSER
```
Comment (quarckster) [inline tox.ini:5]: There are no TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH env vars in gh actions.

**Review by quarckster — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  build-and-publish:
+    name: Build and publish Python 🐍 distributions to PyPI
+    if: startsWith(github.event.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout to master
+        uses: actions/checkout@master
+
+      - name: Setup python
+        uses: actions/setup-python@v1
```
Comment (quarckster) [inline .github/workflows/deploy.yml:20]: Is there a reason to use v1 here? actions/setup-python@v2 is available.

File: .github/workflows/deploy.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      - '*'
+
+jobs:
+  build-and-publish:
+    name: Build and publish Python 🐍 distributions to PyPI
+    if: startsWith(github.event.ref, 'refs/tags')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout to master
+        uses: actions/checkout@master
```
Comment (quarckster) [inline .github/workflows/deploy.yml:17]: Please don't use master versions of actions. There is `actions/checkout@v2`

File: .github/workflows/test_suite.yml [hunk 7]
```diff
@@ -0,0 +1,118 @@
+name: 🕵️ Test suite
+
+on:
+  push:
+  pull_request:
```
Comment (quarckster) [inline .github/workflows/test_suite.yml:5]: Please specify types here:
```
 pull_request:
    types: ["opened", "synchronize", "reopened"]
```

**Review by quarckster — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+          architecture: 'x64'
+
+      - name: Build Package and Check
+        run: |
+          python -m pip install --upgrade setuptools wheel twine
+          python setup.py sdist bdist_wheel
+          python -m twine check dist/*
+
+      - name: Deploy to PyPi
+        uses: pypa/gh-action-pypi-publish@master
```
Comment (quarckster) [inline .github/workflows/deploy.yml:32]: Please use a specific version, I checked the repo and the latest is `v1.3.1`

**Review by quarckster — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 8 above)
Comment (digitronik) [reply .github/workflows/deploy.yml:32]: right latest tag

**Comment by digitronik** [comment]
Thanks, @quarckster for review and suggestions.

**Review by quarckster — APPROVED** [review]

**Pull request merged**